src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs

```diff
@@ -285,6 +285,43 @@ public void SslStream_StreamToStream_Write_ReadByte_Success()
 
         [OuterLoop] // TODO: Issue #11345
         [Fact]
+        public async Task SslStream_StreamToStream_WriteAsync_ReadAsync_Pending_Success()
+        {
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var clientSslStream = new SslStream(clientStream, false, AllowAnyServerCertificate))
+            using (var serverSslStream = new SslStream(serverStream))
+            {
+                bool result = DoHandshake(clientSslStream, serverSslStream);
+                Assert.True(result, "Handshake completed.");
+
+                byte serverReadByte = 0;
+                var readTask = Task.Run(async () =>
+                {
+                    var serverBuffer = new byte[1];
+                    await serverSslStream.ReadAsync(serverBuffer, 0, serverBuffer.Length);
+                    serverReadByte = serverBuffer[0];
+                });
+
+                // Should not hang
+                await serverSslStream.WriteAsync(new byte[] { 1 }, 0, 1);
```

Comment:
There is a race condition in this test, such that it may frequently not test what you want it to test.  You're using Task.Run to schedule the work that calls ReadAsync.  That means that some other thread pool thread needs to process that Task.Run task before this thread goes on to call WriteAsync; otherwise, we're likely to call WriteAsync before that other thread pool thread has gotten around to calling ReadAsync, and we'll end up calling ReadAsync after WriteAsync rather than before.